client/my-sites/upgrades/checkout/transaction-steps-mixin.jsx

```diff
@@ -50,49 +47,21 @@ var TransactionStepsMixin = {
 		this._finishIfLastStep( cart, selectedSite, step );
 	},
 
-	getErrorFromApi: function( errorMessage ) {
-		if ( errorMessage ) {
-			const errorArray = errorMessage.split( /<a href="(.+)">(.+)<\/a>/ );
-
-			if ( errorArray.length === 4 ) { // This assumes we have only one link
-				const errorText1 = errorArray[ 0 ],
-					errorUrl = errorArray[ 1 ],
-					errorLinkText = errorArray[ 2 ],
-					errorText2 = errorArray[ 3 ];
-
-				return (
-					<span>
-						{ errorText1 } <a href={ errorUrl }>{ errorLinkText }</a> { errorText2 }
-					</span>
-				);
-			}
-
-			return errorMessage;
-		}
-
-		return this.translate( 'There was a problem completing the checkout. Please try again.' );
-	},
-
 	_displayNotices: function( cart, step ) {
 		if ( step.error ) {
-			if ( typeof step.error.message === 'object' ) {
-				notices.error( <ValidationErrorList messages={ flatten( values( step.error.message ) ) } /> );
-			} else {
-				notices.error( this.getErrorFromApi( step.error.message ) )
-			}
-
+			upgradesNotices.displayError( step.error );
 			return;
 		}
 
 		switch ( step.name ) {
 			case 'input-validation':
 				if ( ! cartItems.hasFreeTrial( cart ) ) {
-					notices.info( isFree( cart ) ? this.translate( 'Submitting' ) : this.translate( 'Submitting payment' ) );
+					upgradesNotices.displaySubmitting( { isFreeCart: isFree( cart ) } );
 				}
 				break;
 
 			case 'received-wpcom-response':
-				notices.clearNotices( 'notices' );
+				upgradesNotices.clear();
```

Comment:
Well since `upgradesNotices` now handles show upgrades notices I guess it should also handle hiding them. This way we only import `upgradesNotices` and not `notices`